### PR TITLE
Add bonding test for assisted-test-infra

### PR DESCRIPTION
**Pull request by ori-amizur**
/cc @adriengentil 
/cc @osherdp

### CODE REVIEW

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 1]
```diff
@@ -732,6 +732,17 @@ tests:
...
+- as: e2e-metal-assisted-bond
+  optional: true
+  run_if_changed: ^(data/.*)$
+  steps:
+    cluster_profile: packet-assisted
+    env:
+      ASSISTED_CONFIG: |
+        IS_BONDED=true
+        TEST_FUNC=test_networking
+      PACKET_PLAN: m3.large.x86
```
Comment (adriengentil) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:744]:
```suggestion
```
we spawn a 3 masters/2workers cluster, right? In this case the default machine should work

Comment (ori-amizur) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:744]: Actually I copied from STATIC_IP. It has the same configuration.  Do you know why ?

Comment (adriengentil) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:744]: you need to test on OVN and SDN? https://sourcegraph.com/github.com/openshift/assisted-test-infra/-/blob/src/tests/test_e2e_install.py?L21-23

If that's the case then we can let like this

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml [hunk 2]
```diff
@@ -564,6 +564,17 @@ tests:
...
+  cron: 45 03 */1 * 0-5
+  steps:
+    cluster_profile: packet-assisted
+    env:
+      ASSISTED_CONFIG: |
+        IS_BONDED=true
+        TEST_FUNC=test_networking
+      PACKET_PLAN: m3.large.x86
+      TEST_SUITE: full
+    workflow: assisted-baremetal
```
Comment (adriengentil) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:577]:
```suggestion
    workflow: assisted-baremetal
- as: e2e-metal-assisted-bond
  steps:
    cluster_profile: packet-assisted
    env:
      ASSISTED_CONFIG: |
        IS_BONDED=true
        TEST_FUNC=test_networking
      PACKET_PLAN: m3.large.x86
      TEST_SUITE: full
```
let's add a presubmit as well

Comment (ori-amizur) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:577]: OK

**Comment by ori-amizur** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-metal-assisted-bond

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml (hunk 2 above)
Comment (ori-amizur) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:577]: Isn't this presubmit ? https://github.com/openshift/release/blob/178f562bc56ce05717b3ce5de5bfba26d565921e/ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml#L735

**Comment by ori-amizur** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-metal-assisted-bond-periodic

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml (hunk 2 above)
Comment (adriengentil) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:577]: in `assisted-service` repo, not in `assisted-test-infra`

**Comment by ori-amizur** [comment]
/pj-rehearse pull-ci-openshift-assisted-service-master-edge-e2e-metal-assisted-bond

**Comment by ori-amizur** [comment]
/pj-rehearse pull-ci-openshift-assisted-test-infra-master-e2e-metal-assisted-bond

**Comment by ori-amizur** [comment]
/pj-rehearse periodic-ci-openshift-assisted-test-infra-master-e2e-metal-assisted-bond-periodic

**Review by adriengentil — APPROVED** [review]

**Pull request merged**